ojdbc-provider-jackson-oson/src/main/java/oracle/jdbc/provider/oson/OsonFactory.java

```diff
@@ -0,0 +1,402 @@
+   */
+  @Override
+  public JsonParser createParser(char[] content, int offset, int len) throws IOException {
+    IOContext ctxt = _createContext(content, true);
+    Reader r = new CharArrayReader(content, offset, len);
+    if (_inputDecorator != null) {
+      r = _inputDecorator.decorate(ctxt, r);
+    }
+    return _createParser(r, ctxt);
+  }
+
+
+  /**
+   * Creates a OSON parser from a InputStream
+   *
+   * @param in InputStream to use for reading content to parse
+   * @param ctxt I/O context to use for parsing
+   *
+   * @return
+   */
+  @Override
+  public JsonParser _createParser(InputStream in, IOContext ctxt) {
+    return new OsonParser(ctxt, _factoryFeatures, factory.createJsonBinaryParser(in));
+  }
+
+  /**
+   *  Creates a OSON parser from a Reader
+   * @param r Reader to use for reading content to parse
+   * @param ctxt I/O context to use for parsing
+   *
+   * @return
+   */
+  @Override
+  public JsonParser _createParser(Reader r, IOContext ctxt) {
+    
+    return new OsonParser(ctxt, _factoryFeatures, factory.createJsonTextParser(r));
+  }
+
+  /**
+   *  Creates a OSON parser from a char[].
+   *
+   * @param data Buffer that contains content to parse
+   * @param offset Offset to the first character of data to parse
+   * @param len Number of characters within buffer to parse
+   * @param ctxt I/O context to use for parsing
+   * @param recyclable Whether input buffer is recycled by the factory
+   *
+   * @return
+   */
+  @Override
+  public JsonParser _createParser(char[] data, int offset, int len, IOContext ctxt, boolean recyclable) {
+    return new OsonParser(ctxt, _factoryFeatures, 
+        factory.createJsonTextParser(new CharArrayReader(data, offset, len)));
+  }
+
+  /**
+   *  Creates a JSON parser from a byte[]
+   * @param data Buffer that contains content to parse
+   * @param offset Offset to the first character of data to parse
+   * @param len Number of characters within buffer to parse
+   * @param ctxt I/O context to use for parsing
+   *
+   * @return
+   */
+  @Override
+  public JsonParser _createParser(byte[] data, int offset, int len, IOContext ctxt) {
+    return new OsonParser(ctxt, _factoryFeatures, 
+        factory.createJsonBinaryParser(ByteBuffer.wrap(data, offset, len)));
+  }
+
+  /**
+   * Creates a JSON parser from a OracleJsonParser.
+   * @param oParser
+   * @return
+   */
+  public JsonParser createParser(OracleJsonParser oParser) {
+    return new OsonParser(this._createContext(null, false), _factoryFeatures, oParser);
+  }
+
+  /**
+   * Creating OSON parser is not supported for DataInput.
+   * @param input DataInput to use for reading content to parse
+   * @param ctxt I/O context to use for parsing
+   *
+   * @return
+   * @throws IOException
+   */
+  @Override
+  protected JsonParser _createParser(DataInput input, IOContext ctxt) throws IOException {
+    throw new UnsupportedOperationException("Cannot create parser for Data input values");
```

Comment:
DataInput cannot be converted to a type that the OSON parser can read?

Reply:
Our factory does't support to create a parser from DataInput

Reply:
And it can't be wrapped by a Stream or something else?